frontend/packages/dev-console/src/components/buildconfig/sections/EditorField.tsx

```diff
@@ -0,0 +1,49 @@
+import * as React from 'react';
+import { FormGroup } from '@patternfly/react-core';
+import { FormikValues, useFormikContext } from 'formik';
+import MonacoEditor, { MonacoEditorProps } from 'react-monaco-editor';
+
+type EditorFieldProps = {
+  name: string;
+  label?: React.ReactNode;
+  helpText?: React.ReactNode;
+  required?: boolean;
+  isDisabled?: boolean;
+} & MonacoEditorProps;
+
+const EditorField: React.FC<EditorFieldProps> = ({
+  name,
+  label,
+  helpText,
+  required,
+  isDisabled,
+  onChange,
+  ...otherProps
+}) => {
+  const { getFieldMeta, setFieldValue, setFieldTouched } = useFormikContext<FormikValues>();
+  const { error, value } = getFieldMeta<string>(name);
+
+  return (
+    <FormGroup
+      fieldId=""
+      label={label}
+      helperText={helpText}
+      validated={error ? 'error' : 'default'}
+      isRequired={required}
+    >
+      <MonacoEditor
```

Comment:
Would be good to lazy load this component to avoid pulling whole of monaco editor as dependency into main vendor bundle.

Reply:
Does this change the vendor bundle? I need to check how I can test this.

The complete route is currently lazy loaded via `packages/dev-console/src/plugin.tsx`, @invincibleJai noticed below that we should switch to dynamic plugin instead.

I can check this or maybe we can do both as an follow up ticket?

Reply:
If the complete route is lazy loaded it should be fine. But if this was used as a static plugin and imported through plugin.tsx it would create issues.

Reply:
Maybe we can even move this to formik form fields later.